tests/unit/coursier_test.bzl

```diff
@@ -476,6 +477,42 @@ def _get_coursier_cache_or_default_enabled_with_home_dot_coursier_directory_test
 
 get_coursier_cache_or_default_enabled_with_home_dot_coursier_directory_test = add_test(_get_coursier_cache_or_default_enabled_with_home_dot_coursier_directory_test)
 
+def _coursier_credentials_test(ctx):
+    env = unittest.begin(ctx)
+    netrc_content = """
+        machine example.com login foo password bar
+        default login is-ignored
+        machine ignored login missing-password
+        machine missing-password login login
+        machine with-newline
+          login foo
+          password bar
+        machine machine login foo password bar account ignored
+    """
+
+    asserts.equals(
+        env,
+        ["example.com foo:bar",
+         "with-newline foo:bar",
+         "machine foo:bar",
+         ],
+        generate_coursier_credentials_from_netrc(netrc_content)
+    )
+    return unittest.end(env)
+
```

Comment:
Do you mind adding a test for an invalid netrc file since you've covered those cases in your code?

Reply:
I changed the function to use the `read_netrc` helper provided by Bazel, so we are no longer testing parsing of `.netrc` files. This should resolve this thread.